IntegrationAdapters/Controllers/MedicineController.cs

```diff
@@ -19,17 +24,62 @@ public MedicineController()
             this.medicineService = new MedicineService();
         }
         [HttpGet("getMedicineSpecification/{MedicineName}")]
-        public String GetMedicineSpeification(String MedicineName)
+        public IActionResult GetMedicineSpeification(String MedicineName)
         {
             Medicine medicine = medicineService.GetMedicineByName(MedicineName);
             if (medicineService.GetMedicineByName(MedicineName) != null)
```

Comment:
The same method is called here two times? Maybe make a service method DoesMedicineExist(medicineName) that returns true or false